examples/PalmerPenguins.jl

```diff
@@ -0,0 +1,65 @@
+using PalmerPenguins, Plots
+const TABLE = PalmerPenguins.load()
+
+using DataFrames, Random
+df = dropmissing(DataFrame(TABLE))
+df[randperm(nrow(df)), :]
+first(df, 5)
+
+# Binary classification using Bernoulli likelihood - sex
+
+unique(df[:sex])
+
+
+using KernelFunctions, AbstractGPs, GPLikelihoods
+x = RowVecs(Array(df[:, [:bill_length_mm, :bill_depth_mm, :flipper_length_mm, :body_mass_g]]))
+y = [sex=="female" ? true : false for sex in df[:sex]]
+
+scatter(
+    df[:bill_length_mm],
+    df[:bill_depth_mm],
+    df[:flipper_length_mm],
+    group = df[:species],
+    m = (0.5, [:+ :h :star7], 5),
+)
+
+x_train, y_train = x[1:266], y[1:266]
+x_test, y_test = x[267:end], y[267:end]
+
+
+k = Matern52Kernel()
+f = LatentGP(GP(k), BernoulliLikelihood(), 0.001)
+fx = f(x_train)
+logpdf(fx, rand(fx))
+
+
+using EllipticalSliceSampling, Distributions
+function logp(params; x=x_train, y=y_train)
+    kernel = ScaledKernel(
+        KernelFunctions.transform(
+            Matern52Kernel(),
+            ScaleTransform(exp(params[1]))
+        ),
+        exp(params[2])
+    )
+    f = LatentGP(GP(k), BernoulliLikelihood(), 0.001)
+    fx = f(x_train)
+    return logpdf(fx, rand(fx))
```

Comment:
Same as above, these computations should include `y` and not just evaluate a random sample. For the correct log probability here you have to marginalize out `f` (the outputs of the GP). However, for inference it should be sufficient to sample a latent vector (at least that was the case for LGCP that I worked with a while ago).

Reply:
> However, for inference it should be sufficient to sample a latent vector (at least that was the case for LGCP that I worked with a while ago).

Do you mean that I should just consider one sample latent vector like I have done now?

Reply:
I mean the setup here is a bit strange (and I'm not sure if it is correct at all). Usually you would perform Bayesian inference for both the latent vector `f(x_train)` and the parameters. If you are only interested in the posterior distribution of the parameters, preferably you marginalize out the latent vector if possible and only perform inference of the parameters (however, often this is not possible) - currently you use a MC approximation of it but I'm not sure if that's good enough. Additionally, to be able to use elliptical slice sampling, you have to ensure that the prior of the latent vector + the parameters is normally distributed.

In general it can be tricky to work with GPs with non-Gaussian likelihoods (see e.g. https://www.tandfonline.com/doi/pdf/10.1080/00949655.2017.1326117?needAccess=true or https://www.jmlr.org/papers/volume9/nickisch08a/nickisch08a.pdf). Often one also does not perform inference of both the parameters and the latent vector but optimizes the hyperparameters by maximizing the model evidence.

Reply:
Yeah I realize that. A naive way would probably be to fit the latent vectors (f) directly to the outputs (y). But this wouldn't be making use of the Bernoulli likelihood. I think section 3.4 from GPML presents another way to accomplish binary classification. I will look into all and get back to you.

Reply:
@devmotion I have tried to incorporate some of what you said. Could you take a second look and let me know if I am on the right track? Thanks!